src/constructApplications.js

```diff
@@ -146,37 +161,25 @@ function placeLoader(appName, appRoute, loadingPromise) {
     return Promise.all([parcel.mountPromise, loadingPromise]).then(
       ([mountResult, app]) =>
         parcel.unmount().then(() => {
-          if (hidElement) {
-            applicationElement.style.removeProperty("display");
-            if (applicationElement.getAttribute("style") === "") {
-              applicationElement.removeAttribute("style");
-            }
+          if (makeElementVisible) {
+            makeElementVisible();
           }
-
           return app;
         })
     );
   });
 }
 
 function htmlToParcelConfig(str) {
-  const doc = new DOMParser().parseFromString(str, "text/html");
-  let nodes = doc.body.childNodes;
-  let appendedNodes = [];
-
   return {
     bootstrap: () => Promise.resolve(),
     mount: (props) =>
       Promise.resolve().then(() => {
-        nodes.forEach((node) => {
-          appendedNodes.push(props.domElement.appendChild(node.cloneNode()));
-        });
+        props.domElement.innerHTML = str;
```

Comment:
I was doing weird things previously to try to avoid innerHTML for security purposes. But it turns out my implementation was not only buggy but also didn't provide any security (i think). So I switched to innerHTML to fix it.